dlt/destinations/impl/lancedb/lancedb_client.py

```diff
@@ -0,0 +1,567 @@
+        super().update_stored_schema(only_tables, expected_update)
+        applied_update: TSchemaTables = {}
+
+        try:
+            schema_info = self.get_stored_schema_by_hash(self.schema.stored_version_hash)
+        except DestinationUndefinedEntity:
+            schema_info = None
+
+        if schema_info is None:
+            logger.info(
+                f"Schema with hash {self.schema.stored_version_hash} "
+                "not found in the storage. upgrading"
+            )
+            self._execute_schema_update(only_tables)
+        else:
+            logger.info(
+                f"Schema with hash {self.schema.stored_version_hash} "
+                f"inserted at {schema_info.inserted_at} found "
+                "in storage, no upgrade required"
+            )
+        return applied_update
+
+    def _execute_schema_update(self, only_tables: Iterable[str]) -> None:
+        for table_name in only_tables or self.schema.tables:
+            exists = self.table_exists(self.make_qualified_table_name(table_name))
+            if not exists:
+                self.create_table(
+                    self.make_qualified_table_name(table_name),
+                    schema=cast(LanceModel, NullSchema),
+                )
+        self.update_schema_in_storage(self.schema)
+
+    def update_schema_in_storage(self, schema: Schema) -> None:
+        properties = {
+            "version_hash": schema.stored_version_hash,
+            "schema_name": schema.name,
+            "version": schema.version,
+            "engine_version": schema.ENGINE_VERSION,
+            "inserted_at": str(pendulum.now()),
+            "schema": json.dumps(schema.to_dict()),
+        }
+        version_table_name = self.make_qualified_table_name(self.schema.version_table_name)
+        self.create_record(properties, VersionSchema, version_table_name)
+
+    def create_record(
+        self, record: DictStrAny, lancedb_model: TLanceModel, table_name: str
+    ) -> None:
+        """Inserts a record into a LanceDB table without a vector.
+
+        Args:
+            record (DictStrAny): The data to be inserted as payload.
+            table_name (str): The name of the table to insert the record into.
+            lancedb_model (LanceModel): Pydantic model to parse records.
+        """
+        try:
+            tbl = self.db_client.open_table(self.make_qualified_table_name(table_name))
+        except FileNotFoundError:
+            tbl = self.db_client.create_table(self.make_qualified_table_name(table_name))
+        except Exception:
+            raise
+
+        tbl.add(lancedb_model(**record))
+
+    def get_stored_state(self, pipeline_name: str) -> Optional[StateInfo]:
+        """Loads compressed state from destination storage by finding a load ID that was completed."""
+        while True:
+            try:
+                state_table_name = self.make_qualified_table_name(self.schema.state_table_name)
+                state_records = (
```

Comment:
you should order this by load_id descending, is that possible? Check the exact similar thing in weaviate